guides/21_Background_Filter_VideoFX_Processor.md

```diff
@@ -0,0 +1,382 @@
+  - [Using the cross-origin opener policy](#cross-origin-policy)
+  - [SIMD support](#simd-support)
+  - [WebGL2 support](#webgl2-support)
+  - [Content delivery and bandwidth](#delivery-caching-bandwidth)
+  - [Browser compatibility](#filters-browser-compat)
+- [Using a content-security policy](#content-security)
+- [Adding background filters to your application](#add-filters)
+  - [Checking for support before offering a filter](#support-check)
+  - [Creating a VideoFxConfig object](#create-videofxconfig)
+  - [Creating a VideoFxProcessor object](#create-videofxprocessor)
+  - [Creating the VideoTransformDevice object](#create-video-transform)
+  - [Tuning resource utilization](#tuning)
+  - [Starting video input](#start-video-input)
+  - [Stopping video input](#stop-video-input)
+- [Example background filter](#example-bg-filter)
+
+# About using background filters<a id="about-using-background-filters"></a>
+
+## Using the cross\-origin opener policy<a id="cross-origin-policy"></a>
+
+To limit memory usage, the module prefers to use a `SharedArrayBuffer` for processing\. However, this requires that you carefully configure web security\. You must set both of these headers when serving your application HTML:
+
+```
+Cross-Origin-Opener-Policy: same-origin
+Cross-Origin-Embedder-Policy: require-corp
+```
+
+The server must set these because they have no meta\-tag equivalents\. If you don't set these headers, the background filters may use slightly more RAM\.
+
+Background filters can be CPU\-intensive and GPU\-intensive\. Some mobile devices and lower\-specification laptop or desktop computers may not have the power to run the filters along with multiple video streams\.
+
+## SIMD support<a id="simd-support"></a>
+
+Background filters are more efficient in environments that support Single Instruction, Multiple Data (SIMD). The filters use less CPU for a given complexity level when you enable SIMD\. Low\-powered devices running browsers without SIMD support may not run background filters\.
+
+## WebGL2 support<a id="webgl2-support"></a>
+
+The `VideoFxProcessor` object requires browsers that support WebGL2 in order to access the GPU on the client device\.
+
+## Content delivery and bandwidth<a id="delivery-caching-bandwidth"></a>
+
+An Amazon content delivery network loads the machine\-learning\-model files for background filters at runtime\. This provides low\-latency global distribution without the need to serve a full suite of files as part of your application\. However, loading model files can add latency to parts of your application\. To help mitigate that impact, browsers cache the model files indefinitely\. That cache makes subsequent loads significantly faster\. As a best practice, check for supported browsers, then create the background filter resources when users may not notice any latency\. For example, you can download model files while users wait in a lobby, or while they use a device picker\.
```

Comment:
`An Amazon content delivery network loads the machine\-learning\-model files for background filters at runtime when the `.create()` function is invoked.`